crates/kornia-imgproc/src/color/yuv.rs

```diff
@@ -0,0 +1,294 @@
+///
+/// * R: The red channel in the range [0, 255].
+/// * G: The green channel in the range [0, 255].
+/// * B: The blue channel in the range [0, 255].
+///
+/// Precondition: the input image must have 3 channels.
+/// Precondition: the output image must have 3 channels.
+/// Precondition: the input and output images must have the same size.
+///
+/// # Example
+///
+/// ```
+/// use kornia_image::{Image, ImageSize};
+/// use kornia_imgproc::color::rgb_from_yuv;
+///
+/// let image = Image::<f32, 3>::new(
+///     ImageSize {
+///        width: 4,
+///        height: 5,
+///     },
+///     vec![0f32; 4 * 5 * 3],
+/// )
+/// .unwrap();
+///
+/// let mut rgb = Image::<f32, 3>::from_size_val(image.size(), 0.0).unwrap();
+///
+/// rgb_from_yuv(&image, &mut rgb).unwrap();
+///
+/// assert_eq!(rgb.num_channels(), 3);
+/// assert_eq!(rgb.size().width, 4);
+/// assert_eq!(rgb.size().height, 5);
+/// ```
+pub fn rgb_from_yuv(src: &Image<f32, 3>, dst: &mut Image<f32, 3>) -> Result<(), ImageError> {
+    if src.size() != dst.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dst.cols(),
+            dst.rows(),
+        ));
+    }
+
+    // compute the RGB values
+    parallel::par_iter_rows(src, dst, |src_pixel, dst_pixel| {
+        // Transform the input to the range [0, 255]
+        let y = src_pixel[0] * 255.;
+        let u = src_pixel[1] * 255.;
+        let v = src_pixel[2] * 255.;
+
+        let r = y + 1.140 * v;
+        let g = y - 0.396 * u - 0.581 * v;
+        let b = y + 2.029 * u;
+
+        dst_pixel[0] = r;
+        dst_pixel[1] = g;
+        dst_pixel[2] = b;
+    });
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use kornia_image::{Image, ImageError, ImageSize};
+    use num_traits::Pow;
+    const RGB_TEST_DATA: [f32; 18] = [
+        0.0, 128.0, 255.0, 255.0, 128.0, 0.0, 128.0, 255.0, 0.0, 255.0, 0.0, 128.0, 0.0, 128.0,
+        255.0, 255.0, 128.0, 0.0,
+    ];
+    // corresponding YUV values to the RGB_TEST_DATA
+    const YUV_TEST_DATA: [f32; 18] = [
+        0.4087, 0.2909, -0.3585, 0.5937, -0.2921, 0.3565, 0.7371, -0.3628, -0.2063, 0.3562, 0.0719,
+        0.5648, 0.4087, 0.2909, -0.3585, 0.5937, -0.2921, 0.3565,
+    ];
+    const RGB_FROM_YUV_DATA: [f32; 18] = [
+        0.002548, 127.956985, 254.7287, 255.02805, 128.0725, 0.262405, 127.98908, 255.16042,
+        0.249588, 255.01837, -0.107399, 128.03171, 0.002548, 127.956985, 254.7287, 255.02805,
+        128.0725, 0.262405,
+    ];
+    const WIDTH: usize = 2;
+    const HEIGHT: usize = 3;
+    #[test]
+    fn yuv_from_rgb() -> Result<(), ImageError> {
+        let image = Image::<f32, 3>::new(
+            ImageSize {
+                width: WIDTH,
+                height: HEIGHT,
+            },
+            RGB_TEST_DATA.to_vec(),
+        )?;
+        let expected = YUV_TEST_DATA;
+
+        let mut yuv = Image::<f32, 3>::from_size_val(image.size(), 0.0)?;
```

Comment:
```suggestion
        let mut yuv = Image::from_size_val(image.size(), 0.0)?;
```
the compiler should be able to resolve that